### PR TITLE
admin: refine admin check.

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
refine admin check.

### What is changed and how it works?
Admin check uses `rowDecoder` to decode data and evaluates generated column.

### Check List 

Code changes

 - Has exported function/method change

Related changes

No need to cherry pick.

### CODE REVIEW

**Comment by shenli** [comment]
Is this PR improve the performance?

**Comment by crazycs520** [comment]
@shenli 
1. For index on generated columns, may have a little performance improvement. This PR will reduce memory allocation when evaluates virtual generated column.
2. Normally, this PR have not improve the performance, But may make the code more clearer.

File: util/admin/admin.go [hunk 1]
```diff
@@ -582,34 +606,13 @@ func rowWithCols(sessCtx sessionctx.Context, txn kv.Retriever, t table.Table, h
...
-		for _, c := range t.Cols() {
-			if c.State != model.StatePublic {
-				continue
-			}
-			colTps[c.ID] = &c.FieldType
-		}
 	}
 
-	rowMap, err := tablecodec.DecodeRow(value, colTps, sessCtx.GetSessionVars().Location())
+	rowMap, err := rowDecoder.DecodeAndEvalRowWithMap(sessCtx, value, sessCtx.GetSessionVars().Location(), time.UTC, nil)
```
Comment (winkyao) [inline util/admin/admin.go]: Why the time.Location change to time.UTC, not sessCtx.GetSessionVars().Location()?

Comment (crazycs520) [reply util/admin/admin.go]: There is not change.
`sessCtx.GetSessionVars().Location()` is for `tablecodec.DecodeRow` to decode data.
`time.UTC` is for evaluated virtual generated column.

File: planner/core/planbuilder.go [hunk 2]
```diff
@@ -548,8 +548,7 @@ func (b *planBuilder) buildAdminCheckTable(as *ast.AdminStmt) (*CheckTable, erro
 				return nil, errors.Trace(err)
 			}
 			expr = expression.BuildCastFunction(b.ctx, expr, colExpr.GetType())
-			genColumnName := model.GetTableColumnID(tableInfo, column.ColumnInfo)
-			p.GenExprs[genColumnName] = expr
+			p.GenExprs[model.TableColumnID{TableID: tableInfo.ID, ColumnID: column.ColumnInfo.ID}] = expr
```
Comment (ciscoxll) [inline planner/core/planbuilder.go]: Will there be a problem here?

Comment (crazycs520) [reply planner/core/planbuilder.go]: what problem?

**Pull request merged**